internal/rpc/starknet/starknet.go

```diff
@@ -174,6 +174,19 @@ func (s *StarkNetRpc) GetBlockTransactionCount(blockId *BlockId) (any, error) {
 	return b.TxCount, nil
 }
 
+func (s *StarkNetRpc) GetNonce(blockId *BlockId, address *RpcFelt) (any, error) {
+	b, err := getBlockById(blockId, s.blockManager, s.logger)
+	if err != nil {
+		return nil, err
+	}
+	_state := state.New(s.stateManager, b.NewRoot)
+	nonce, err := _state.GetNonce(address.Felt())
+	if err != nil {
+		return nil, err
+	}
+	return nonce, nil
```

Comment:
Use the `felt.Hex0x()` to return felts in hex format, otherwise, the decimal representation is sent